HerPortal.ManagementShell/AdminAction.cs

```diff
@@ -32,161 +35,87 @@ public AdminAction(IDatabaseOperation dbOperation, IOutputProvider outputProvide
+        var consortiaToAdd = dbOperation.GetConsortia(consortiumCodes ?? Array.Empty<string>());
+
+        dbOperation.CreateUserOrLogError(userEmailAddress, lasToAdd, consortiaToAdd);
     }
 
-    public void TryRemoveUser(User? user)
+    public void RemoveUser(User user)
     {
-        if (user == null)
-        {
-            outputProvider.Output("User not found");
-            return;
-        }
-
-        var deletionConfirmation = outputProvider.Confirm(
-            $"Attention! This will delete user {user.EmailAddress} and all associated rows from the database. Are you sure you want to commit this transaction? (y/n)");
-        if (!deletionConfirmation)
-        {
-            return;
-        }
-
         dbOperation.RemoveUserOrLogError(user);
     }
-    private void AddLas(User? user, string[]? custodianCodes)
+
+    public void AddLas(User user, IEnumerable<string> custodianCodes)
     {
-        if (user == null)
-        {
-            outputProvider.Output("User not found");
-            return;
-        }
-
-        if (custodianCodes == null || custodianCodes.Length < 1)
-        {
-            outputProvider.Output("Please specify custodian codes to add to user");
-            return;
-        }
-
-        var lasToAdd = dbOperation.GetLas(custodianCodes);
+        var filteredCustodianCodes = custodianCodes
+            .Where(custodianCode => !CustodianCodeIsInOwnedConsortium(user, custodianCode))
+            .ToList();
+
+        var lasToAdd = dbOperation.GetLas(filteredCustodianCodes);
+
         dbOperation.AddLasToUser(user, lasToAdd);
     }
 
-    public void RemoveLas(User? user, string[]? custodianCodes)
+    public void RemoveLas(User user, IReadOnlyCollection<string> custodianCodes)
     {
-        if (user == null)
-        {
-            outputProvider.Output("User not found");
-            return;
-        }
-
-        if (custodianCodes == null || custodianCodes.Length < 1)
-        {
-            outputProvider.Output("Please specify custodian codes to remove from user");
-            return;
-        }
-
-        var userConfirmation = ConfirmCustodianCodes(user.EmailAddress, custodianCodes);
-        if (!userConfirmation)
-        {
-            return;
-        }
-
         var lasToRemove = user.LocalAuthorities.Where(la => custodianCodes.Contains(la.CustodianCode)).ToList();
         var missingCodes = custodianCodes.Where(code => !lasToRemove.Any(la => la.CustodianCode.Equals(code))).ToList();
         if (missingCodes.Count > 0)
-        {
-            outputProvider.Output($"Could not find LAs attached to {user.EmailAddress} for the following codes: {string.Join(", ", missingCodes)}. Please check your inputs and try again.");
-            return;
-        }
+            throw new KeyNotFoundException(
+                $"Could not find LAs attached to {user.EmailAddress} for the following codes: {string.Join(", ", missingCodes)}. Please check your inputs and try again.");
```

Comment:
I want to nitpick a bit here: I don't think this is the right exception to be throwing. `KeyNotFoundException` is specifically for when we're looking things up by key in a collection (e.g. dictionary) in the code. A `KeyNotFoundException` would make me think we're looking in one of our dictionaries, for example. We're not really doing that here - this is based on comparing with the results of a query.

Reply:
(also do we need to be throwing an exception at all now? vs what it was doing before?)

Reply:
in this case I think so; AdminAction doesn't do printing any more so the error can't be handled here

Reply:
in any case replaced the KeyNotFoundException with a custom CommandException